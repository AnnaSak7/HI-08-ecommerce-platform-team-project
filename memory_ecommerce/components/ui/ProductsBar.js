import * as React from "react";
import Tabs from "@mui/material/Tabs";
import Tab from "@mui/material/Tab";
import Box from "@mui/material/Box";
import { useRouter } from "next/router";

export default function ProductsBar(props) {
  const [value, setValue] = React.useState("1");
  const router = useRouter();

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  const showDetailsHandler = (path) => {
    router.push(`/${path}`);
    console.log("value is ", path);
  };

  return (
    <Box sx={{ width: "100%" }}>
      <Tabs
        //variant="fullWidth"
        centered
        value={value}
        onChange={handleChange}
        textColor="secondary"
        indicatorColor="secondary"
        aria-label="secondary tabs example"
      >
        <Tab
          id="1"
          value="1"
          label="Lavish trip"
          onClick={(e) => showDetailsHandler("/626670ddf92bdf3fb20675aa")}
        />
        <Tab
          id="2"
          value="2"
          label="Long night out"
          onClick={(e) => showDetailsHandler("/6266758df92bdf3fb20675ac")}
        />
        <Tab
          id="3"
          value="3"
          label="Top of the world"
          onClick={(e) => showDetailsHandler("/62666fd2f92bdf3fb20675a9")}
        />
        <Tab
          id="4"
          value="4"
          label="Butterflies in the stomach"
          onClick={(e) => showDetailsHandler("/62666fd2f92bdf3fb20675a9")}
        />
        <Tab
          id="5"
          value="5"
          label="Pearl white beaches"
          onClick={(e) => showDetailsHandler("/62667676f92bdf3fb20675ad")}
        />
      </Tabs>
    </Box>
  );
}
