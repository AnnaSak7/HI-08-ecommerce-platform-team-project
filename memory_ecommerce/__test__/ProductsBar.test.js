import React from "react";
import { render } from "@testing-library/react";
import ProductsBar from "../components/ui/ProductsBar";
import Tabs from "@mui/material/Tabs";
import Tab from "@mui/material/Tab";
import Box from "@mui/material/Box";

// import { useRouter } from "next/router";

// jest.mock("next/router");

// TEST========================================
test("<Tab /> matches snapshot", () => {
  let value;

  const component = render(
    <Box sx={{ width: "100%" }}>
      <Tabs
        centered
        value={value}
        onChange={(event, newValue) => {
          setValue(newValue);
        }}
      >
        <Tab
          id="1"
          value="1"
          label="Lavish trip"
          onClick={(e) => showDetailsHandler("/626670ddf92bdf3fb20675aa")}
        />
        <Tab
          id="2"
          value="2"
          label="Long night out"
          onClick={(e) => showDetailsHandler("/6266758df92bdf3fb20675ac")}
        />
        <Tab
          id="3"
          value="3"
          label="Top of the world"
          onClick={(e) => showDetailsHandler("/62666fd2f92bdf3fb20675a9")}
        />
        <Tab
          id="4"
          value="4"
          label="Butterflies in the stomach"
          onClick={(e) => showDetailsHandler("/626674def92bdf3fb20675ab")}
        />
        <Tab
          id="5"
          value="5"
          label="Pearl white beaches"
          onClick={(e) => showDetailsHandler("/62667676f92bdf3fb20675ad")}
        />
      </Tabs>
    </Box>
  );
  expect(component.container).toMatchSnapshot();
});

// TEST========================================
test("<Tab /> renders without crashing", () => {
  let value;
  const div = document.createElement("div");
  render(
    <Box sx={{ width: "100%" }}>
      <Tabs
        centered
        value={value}
        onChange={(event, newValue) => {
          setValue(newValue);
        }}
      >
        <Tab
          id="1"
          value="1"
          label="Lavish trip"
          onClick={(e) => showDetailsHandler("/626670ddf92bdf3fb20675aa")}
        />
        <Tab
          id="2"
          value="2"
          label="Long night out"
          onClick={(e) => showDetailsHandler("/6266758df92bdf3fb20675ac")}
        />
        <Tab
          id="3"
          value="3"
          label="Top of the world"
          onClick={(e) => showDetailsHandler("/62666fd2f92bdf3fb20675a9")}
        />
        <Tab
          id="4"
          value="4"
          label="Butterflies in the stomach"
          onClick={(e) => showDetailsHandler("/626674def92bdf3fb20675ab")}
        />
        <Tab
          id="5"
          value="5"
          label="Pearl white beaches"
          onClick={(e) => showDetailsHandler("/62667676f92bdf3fb20675ad")}
        />
      </Tabs>
    </Box>,
    div
  );
});

// TEST========================================
test("Defaults to the first tab's content", () => {
  let value;
  const div = document.createElement("div");
  render(
    <Box sx={{ width: "100%" }}>
      <Tabs
        centered
        value={value}
        onChange={(event, newValue) => {
          setValue(newValue);
        }}
      >
        <Tab
          id="1"
          value="1"
          label="Lavish trip"
          onClick={(e) => showDetailsHandler("/626670ddf92bdf3fb20675aa")}
        />
        <Tab
          id="2"
          value="2"
          label="Long night out"
          onClick={(e) => showDetailsHandler("/6266758df92bdf3fb20675ac")}
        />
        <Tab
          id="3"
          value="3"
          label="Top of the world"
          onClick={(e) => showDetailsHandler("/62666fd2f92bdf3fb20675a9")}
        />
        <Tab
          id="4"
          value="4"
          label="Butterflies in the stomach"
          onClick={(e) => showDetailsHandler("/626674def92bdf3fb20675ab")}
        />
        <Tab
          id="5"
          value="5"
          label="Pearl white beaches"
          onClick={(e) => showDetailsHandler("/62667676f92bdf3fb20675ad")}
        />
      </Tabs>
    </Box>
  );
});

// TEST========================================
test("Can set a different tab as the initial state", () => {});

// TEST========================================
test("Click from one tab to the next", () => {});
