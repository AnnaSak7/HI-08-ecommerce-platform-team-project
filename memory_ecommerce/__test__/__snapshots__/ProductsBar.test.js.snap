// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiTabs-root css-1ujnqem-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            id="1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Lavish trip
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            id="2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Long night out
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            id="3"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Top of the world
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            id="4"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Butterflies in the stomach
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            id="5"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Pearl white beaches
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;
